File: C#/CreationalPatterns/CreationalPatterns.Builder/Building .cs
using System;

namespace CreationalPatterns.Builder
{
    internal class Building
    {
        private readonly Builder builder;
        private readonly Builder builder2;

        public Building()
        {
            builder = new FirstHouse();
        }

        public Building(String s)
        {
            builder2 = new SecondHouse();
        }

        public House BuildHouse()
        {
            return builder.BuildHouse();
        }

        public House BuildHouse2()
        {
            return builder2.BuildHouse();
        }
    }
}

File: Java/creational_patterns/src/Builder/Building.java
package Builder;

public class Building {
    private Builder builder;
    private Builder builder2;

    public Building() {
        builder = new FirstHouse();
    }

    public Building(String s) {
        builder2 = new SecondHouse();
    }

    public House buildHouse() {
        return builder.builHouse();
    }

    public House buildHouse2() {
        return builder2.builHouse();
    }
}
